### PR TITLE
Add Seccomp Option

### PR DESCRIPTION
**Pull request by soharaki**
When running `nerdctl run`, we can use the seccomp flg.
Referred to "Docker's security option: seccomp", not the ctr commands.
#11

Signed-off-by: Sho Haraki <endurance.test.20380119@gmail.com>

### CODE REVIEW

File: run.go [hunk 1]
```diff
@@ -77,6 +77,10 @@ var runCommand = &cli.Command{
 			Usage: "Set custom DNS servers (only meaningful for \"bridge\" network)",
 			Value: cli.NewStringSlice("8.8.8.8", "1.1.1.1"),
 		},
+		&cli.StringFlag{
```
Comment (soharaki) [inline run.go:80]: I wasn't sure whether to add this option to `run.go` or `main.go`.  And I selected `run.go` .
In the containerd project, it is `containerd/cmd/ctr/commands/commands.go`, 
but it is only used as an option for `ctr run`.

File: run.go [hunk 2]
```diff
@@ -153,6 +156,17 @@ func runAction(clicontext *cli.Context) error {
 		return errors.Errorf("unknown network %q", netstr)
 	}
 
+	seccompProfile := clicontext.String("seccomp")
+	if seccompProfile != "" {
+		if seccompProfile != "unconfined" {
```
Comment (soharaki) [inline run.go:161]: The `seccomp` and `unconfined` come from Docker options.

File: run.go [hunk 3]
```diff
@@ -153,6 +156,17 @@ func runAction(clicontext *cli.Context) error {
...
+			opts = append(opts, seccomp.WithProfile(seccompProfile))
+		} else {
+			// If passing "unconfined" to run a container without the default seccomp profile.
+		}
+	} else {
+		opts = append(opts, seccomp.WithDefaultProfile())
+	}
+
```
Comment (soharaki) [inline run.go:169]: This is how it works:

```console
# containerd's seccomp defaultProfile
$ sudo nerdctl run -it --seccomp="" alpine sh
/ # unshare
unshare: unshare(0x0): Operation not permitted

# no seccomp
$ sudo nerdctl run -it --seccomp="unconfined" alpine sh
/ # unshare --user
HOST:~$

# Enable custom seccomp
$ sudo nerdctl run -it --seccomp=_output/default.json alpine sh
/ # unshare
HOST:~$
```
refs: https://github.com/moby/moby/blob/f7bcb02f677257a5623dcb561f003e96d5f3ea3f/profiles/seccomp/default.json

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
The CLI syntax needs to be compatible with Docker, please see https://docs.docker.com/engine/security/seccomp/

**Comment by AkihiroSuda** [comment]
Also, could you use your real name for signing off the commit? Thanks

File: run.go [hunk 4]
```diff
@@ -153,6 +156,17 @@ func runAction(clicontext *cli.Context) error {
 		return errors.Errorf("unknown network %q", netstr)
 	}
 
+	seccompProfile := clicontext.String("seccomp")
+	if seccompProfile != "" {
+		if seccompProfile != "unconfined" {
+			opts = append(opts, seccomp.WithProfile(seccompProfile))
+		} else {
```
Comment (AkihiroSuda) [inline run.go:163]: This empty else{} block is unneeded

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

File: run.go (hunk 1 above)
Comment (AkihiroSuda) [reply run.go:80]: This should be in run.go, but the flag syntax needs to be `--security-opt seccomp=<PROFILE>`

File: run.go [hunk 5]
```diff
@@ -77,6 +79,10 @@ var runCommand = &cli.Command{
 			Usage: "Set custom DNS servers (only meaningful for \"bridge\" network)",
 			Value: cli.NewStringSlice("8.8.8.8", "1.1.1.1"),
 		},
+		&cli.StringFlag{
```
Comment (AkihiroSuda) [inline run.go:82]: cli.StringSliceFlag

File: run.go [hunk 6]
```diff
@@ -153,6 +158,25 @@ func runAction(clicontext *cli.Context) error {
 		return errors.Errorf("unknown network %q", netstr)
 	}
 
+	securityOpt := clicontext.String("security-opt")
```
Comment (AkihiroSuda) [inline run.go:161]: `clicontext.StringSlice`

**Comment by soharaki** [comment]
I tried to matche the behavior of `nerdctl` to docker-cli as closely as possible.
However, the behavior is different when not setting any arguments.

```console
$ docker container run -it --security-opt "" alpine sh
docker: Invalid --security-opt: "".
See 'docker run --help'.

$ sudo nerdctl run -it --security-opt "" alpine sh
/ #
$  sudo nerdctl run -it --security-opt "seccomp=" alpine sh
/ # #the same above
```

File: run.go [hunk 7]
```diff
@@ -153,6 +158,25 @@ func runAction(clicontext *cli.Context) error {
...
+	if securityOpt != "" {
+		securityOpts := regexp.MustCompile("[\\,\\s]+").Split(securityOpt, -1)
```
Comment (AkihiroSuda) [inline run.go:163]: No need to use regexp

File: run.go [hunk 8]
```diff
@@ -153,6 +158,25 @@ func runAction(clicontext *cli.Context) error {
...
+		securityOptsMaps := ConvertKVStringsToMap(securityOpts)
+		dictSecurityOpts := map[string]string{
+			"seccomp": "allow",
```
Comment (AkihiroSuda) [inline run.go:166]: unused value

File: run.go (hunk 6 above)
Comment (soharaki) [reply run.go:161]: I've been thinking about that, but ` --security-opt "seccomp=hoge.json,apparmor=foo.profile"`

Comment (AkihiroSuda) [reply run.go:161]: Docker doesn't support such syntax either

Comment (soharaki) [reply run.go:161]: I supposed that docker are the docker seems to be able to handle the parser in the same way.

Comment (soharaki) [reply run.go:161]: >Docker doesn't support such syntax either
Oh, I see.

Comment (AkihiroSuda) [reply run.go:161]: `docker run --security-opt seccomp=hoge.json --security-opt apparmor=foo.profile" is the correct form

**Comment by soharaki** [comment]
I fixed it.

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**